tools/install-powershell.ps1

```diff
@@ -79,6 +79,68 @@ Function Remove-Destination([string] $Destination) {
     }
 }
 
+Function IsPathInRegistryPath([string] $Path) {
+    #Remove ending DirectorySeparatorChar for comparison purposes
+    $Path = [System.Environment]::ExpandEnvironmentVariables([regex]::Replace($Path, '\\;$|\\$', ''));
+    $InstalledPaths = @()
+    #Possible gotcha: don't use Get-ItemPropertyValue (see https://github.com/PowerShell/PowerShell/issues/5906)
+    #Possible gotcha: coerce the (possibly null) property to string before calling ExpandEnvironmentVariables
+    [string] $HKCURegKeyPath = Get-ItemProperty -Path 'HKCU:\Environment' -Name 'PATH' -ErrorAction SilentlyContinue
+    $InstalledPaths += @([System.Environment]::ExpandEnvironmentVariables(($HKCURegKeyPath)).Split(';'))
+    [string] $HKLMRegKeyPath = Get-ItemProperty -Path 'HKLM:\SYSTEM\CurrentControlSet\Control\Session Manager\Environment' -Name 'PATH' -ErrorAction SilentlyContinue
```

Comment:
Please use the environment API to get the environment:
```suggestion
    [string] $HKLMRegKeyPath = [System.Environment]::GetEnvironmentVariable('Path','Machine')
```

Reply:
done in https://github.com/PowerShell/PowerShell/pull/8081/commits/2fe05aa5d07c220ac077c1fc52ef4231aad66286